state.go

```diff
@@ -478,6 +481,15 @@ func (s *State) applyUpdateProposal(target LeafIndex, update *UpdateProposal) er
 		return fmt.Errorf("mls.state: Invalid kp")
 	}
 
+	currKP, ok := s.Tree.KeyPackage(target)
+	if !ok {
+		return fmt.Errorf("mls.state: Attempt to update an empty leaf")
+	}
+
+	if !update.KeyPackage.Credential.Equals(currKP.Credential) {
+		s.NewCredentials[target] = true
+	}
+
```

Comment:
i wonder if we update the bool map after we perform s.Tree.UpdateLeaf ? aren't we checking the updated credentil before doing the update the way it is coded now ?

Reply:
We are not checking the credential before doing the update.  Credential checking here is lazy, after all the changes have been done.

Reply:
Whether that's a good idea, we should discuss later :)